python/python/tests/test_balanced.py

```diff
@@ -58,6 +58,32 @@ def balanced_dataset(tmp_path, big_val):
     )
 
 
+def test_write_fragments(balanced_dataset, tmp_path):
+    tbl = balanced_dataset._take_rows(range(10))
+    transaction = lance.fragment.write_fragments(
+        tbl,
+        tmp_path / "ds",
+        enable_move_stable_row_ids=True,
+        return_transaction=True,
+    )
+    operation = lance.LanceOperation.Overwrite(
+        transaction.operation.new_schema, transaction.operation.fragments
+    )
+    blob_operation = lance.LanceOperation.Overwrite(
+        transaction.blobs_op.new_schema, transaction.blobs_op.fragments
+    )
+
+    ds = lance.LanceDataset.commit(
```

Comment:
This should be done in a future PR but maybe (now that we can serialize `Transaction`) we need a `LanceDataset.commit_transaction` that can just take in a transaction object instead of requiring us to pass in all the pieces.